core/src/test/java/hudson/UtilTest.java

```diff
@@ -781,4 +781,29 @@ public void testPermissionsToMode() throws Exception {
         assertEquals(0000, Util.permissionsToMode(PosixFilePermissions.fromString("---------")));
     }
 
+    @Test
+    public void testDifferenceDays() {
+        Date may_6_10am = new Date(2018 - 1900, 4, 6, 10, 0);
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#constructor.summary

> Date(int year, int month, int date, int hrs, int min, int sec)
> **Deprecated.**
>As of JDK version 1.1, replaced by Calendar.set(year + 1900, month, date, hrs, min, sec) or GregorianCalendar(year + 1900, month, date, hrs, min, sec).

Reply:
In general I do not care really for test code to use deprecated constructors like that because it's not the code that is tested but only the tools we use to test the real things.

As you pointed out this code and take time to retrieve the documentation, I imagine it's important for you, so I changed it 👍